### PR TITLE
feat(api): add eth_getRawTransactionByHash JSON-RPC method

### LINKED ISSUE #10477: Implement eth_getRawTransactionByHash RPC method
Besu does not implement eth_getRawTransactionByHash, which returns the RLP-encoded raw bytes of a transaction by hash. This method is implemented by Geth, Nethermind, Erigon, and Reth, and improves interoperability for tooling that calls it directly rather than
  going through eth_getTransactionByHash and re-encoding.

  Expected behaviour

`  eth_getRawTransactionByHash(txHash) → "0x<rlp-encoded transaction>"
`
  Returns null if the transaction is not found (consistent with eth_getTransactionByHash).

  Besu already implements debug_getRawTransaction (DebugGetRawTransaction.java), which does exactly this. 

  The implementation also requires:
  1. Adding ETH_GET_RAW_TRANSACTION_BY_HASH("eth_getRawTransactionByHash") to RpcMethod.java
  2. Registering the new method in JsonRpcMethodsFactory (or wherever debug_getRawTransaction is registered) under the ETH API group
  3. Adding a unit test (modelled on DebugGetRawTransactionTest)

  References

  - Geth: eth_getRawTransactionByHash in internal/ethapi/api.go
  - Cross-client RPC method analysis: https://hackmd.io/@bomanaps/rJXLPhYRWl

### PR DESCRIPTION
**Pull request by DhroovSankla**
### Description
This PR implements the `eth_getRawTransactionByHash` JSON-RPC method. This method returns the raw RLP-encoded bytes of a transaction by its hash, improving compatibility and interoperability with tools that call it directly (such as Geth, Nethermind, Erigon, and Reth) rather than having to call `eth_getTransactionByHash` and re-encode.

It leverages and refactors the existing transaction serialization logic used in `debug_getRawTransaction`.

### Changes
- **`RpcMethod.java`**: Added the `ETH_GET_RAW_TRANSACTION_BY_HASH` enum value.
- **`RawTransactionEncoder.java`**: Extracted the raw transaction encoding logic to a shared helper utilizing the modern `TransactionEncoder.encodeOpaqueBytes(...)` from the `main` branch.
- **`DebugGetRawTransaction.java`**: Refactored to use the shared `RawTransactionEncoder`.
- **`EthGetRawTransactionByHash.java`**: Added the new JSON-RPC method implementation querying the transaction pool and blockchain.
- **`EthJsonRpcMethods.java`**: Registered the new method under the `ETH` API group.
- **`EthGetRawTransactionByHashTest.java`**: Added comprehensive unit tests covering EIP-2718 typed transaction decoding, pending transactions, mined transactions, and invalid parameter edge cases.
- **`JsonRpcMethodsFactoryTest.java`**: Verified method exposure.

### Fixes
Fixes #10477

### CODE REVIEW

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetRawTransactionByHashTest.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+  @Test
+  void shouldReturnPendingTransactionRlpWhenTransactionExistsInTransactionPool() {
+    final Transaction transaction =
+        TransactionDecoder.decodeOpaqueBytes(
+            Bytes.fromHexString(VALID_TRANSACTION), EncodingContext.BLOCK_BODY);
+
+    when(transactionPool.getTransactionByHash(transaction.getHash()))
+        .thenReturn(Optional.of(transaction));
+    verifyNoInteractions(blockchainQueries);
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetRawTransactionByHashTest.java]: verifyNoInteractions called before method.response() — test is vacuous

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/RawTransactionEncoder.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.encoding.EncodingContext;
+import org.hyperledger.besu.ethereum.core.encoding.TransactionEncoder;
+
+final class RawTransactionEncoder {
+
+  private RawTransactionEncoder() {}
+
+  static String toRawString(final Transaction transaction) {
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/RawTransactionEncoder.java]: I think adding a new class for this is overkill - a better home would be as a static method on TransactionEncoder itself, and then inline the call rather than call out to this class

**Review by macfarla — COMMENTED** [review]
overall LGTM. Needs a changelog entry added.

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetRawTransactionByHashTest.java (hunk 1 above)
Comment (DhroovSankla) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetRawTransactionByHashTest.java]: Removed the premature verifyNoInteractions call before method.response()

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/RawTransactionEncoder.java (hunk 2 above)
Comment (DhroovSankla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/RawTransactionEncoder.java]: Inlined the TransactionEncoder.encodeOpaqueBytes(..., EncodingContext.BLOCK_BODY).toHexString() call directly and removed RawTransactionEncoder.java

**Review by macfarla — APPROVED** [review]
